app/queries/reasons_for_rejection_applications_query.rb

```diff
@@ -0,0 +1,31 @@
+class ReasonsForRejectionApplicationsQuery
+  attr_accessor :filters
+
+  def initialize(filters)
+    self.filters = filters
+  end
+
+  def call
+    application_choices = ApplicationChoice.where.not(structured_rejection_reasons: nil)
+
+    apply_filters(application_choices)
+  end
+
+private
+
+  def apply_filters(application_choices)
+    filters.each do |key, value|
+      if key =~ /_y_n$/
+        application_choices = application_choices.where(
+          "application_choices.structured_rejection_reasons->>'#{key}' = '#{value}'",
```

Comment:
This just accesses the blob vs accessing the attr right?

Reply:
It's a raw SQL condition on a property inside the JSONB column so it should translate to something like `WHERE application_choices.structured_rejection_reasons->>'qualifications_y_n' = 'Yes'` rather than loading the JSONB value into the model and then filtering in memory. Postgresql syntax for this stuff is a bit odd but it's quite powerful.